editor/components/TableSettingsMenu.tsx

```diff
@@ -0,0 +1,76 @@
+import { Dispatch, SetStateAction, useCallback } from "react";
+import Dropdown from "react-bootstrap/Dropdown";
+import Form from "react-bootstrap/Form";
+import { ColumnVisibilitySetting } from "@/types/types";
+import { FaGear } from "react-icons/fa6";
+
+interface TableSettingsMenuProps {
+  columnVisbilitySettings: ColumnVisibilitySetting[];
+  setColumnVisbilitySettings: Dispatch<
+    SetStateAction<ColumnVisibilitySetting[]>
+  >;
+}
+
+/**
+ * Table component that controls column visibility
+ */
+export default function TableSettingsMenu({
+  columnVisbilitySettings,
+  setColumnVisbilitySettings,
+}: TableSettingsMenuProps) {
+  const handleUpdateColVisibility = useCallback(
+    (columns: ColumnVisibilitySetting[], path: string) => {
+      const updatedColVisibilitySettings = columns.map((col) => {
+        if (col.path !== path) {
+          return col;
+        }
+        return { ...col, isVisible: !col.isVisible };
+      });
+      // do nothing if this will result in all columns being invisible
+      const willAllColumnsBeInivisible = updatedColVisibilitySettings.every(
+        (col) => col.isVisible === false
+      );
+      if (!willAllColumnsBeInivisible) {
+        setColumnVisbilitySettings(updatedColVisibilitySettings);
+      }
+    },
+    [setColumnVisbilitySettings]
+  );
+
+  return (
+    <Dropdown>
+      <Dropdown.Toggle
+        variant="outline-primary"
+        className="border-0 hide-toggle"
+        id="column-visibility-picker"
+      >
+        <FaGear />
+      </Dropdown.Toggle>
+
+      <Dropdown.Menu style={{ maxHeight: "50vh", overflowY: "auto" }}>
```

Comment:
future enhancement, it'd be nice to add a search input to filter columns, as well as maybe toggles for all/none/reset columns.